utils/service_discovery/abstract_config_store.py

```diff
@@ -98,11 +99,12 @@ def _populate_identifier_to_checks(self):
 
         return identifier_to_checks
 
-    def _get_kube_config(self, identifier, kube_annotations):
+    def _get_kube_config(self, identifier, kube_annotations, kube_container_name):
         try:
-            check_names = json.loads(kube_annotations[KUBE_ANNOTATION_PREFIX + CHECK_NAMES])
-            init_config_tpls = json.loads(kube_annotations[KUBE_ANNOTATION_PREFIX + INIT_CONFIGS])
-            instance_tpls = json.loads(kube_annotations[KUBE_ANNOTATION_PREFIX + INSTANCES])
+            prefix = '{}/{}.'.format(KUBE_ANNOTATION_PREFIX, kube_container_name)
```

Comment:
HK - I think you might have forgotten some of the context from above. The main point of doing this PR is so we instantiate the checks above once per pod rather than once per container (since pods can have a few containers and you can't annotate containers directly in kubernetes; this was previously creating N instances of the checks above where N=containers in a pod).